gobgp client bug: add route-server-client neighbor

when add a route-server-client neighbor, there is a bug:

`
gobgp -p 50052 -d neighbor add 127.0.0.4 as 65001 route-server-client
`

output: 

`
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x14d4b55]

goroutine 1 [running]:
main.modNeighbor.func3(0xc000118000)
	/Users/liuxiang/go/pkg/mod/github.com/osrg/gobgp@v0.0.0-20211001064702-91b91278600d/cmd/gobgp/neighbor.go:1276 +0x2b5
main.modNeighbor({0x16232a0, 0x3}, {0xc0002aaa10, 0x4, 0x7})
	/Users/liuxiang/go/pkg/mod/github.com/osrg/gobgp@v0.0.0-20211001064702-91b91278600d/cmd/gobgp/neighbor.go:1318 +0x5dc
main.newNeighborCmd.func7(0xc0002b4a00, {0xc0002aaa10, 0x7, 0x7})
	/Users/liuxiang/go/pkg/mod/github.com/osrg/gobgp@v0.0.0-20211001064702-91b91278600d/cmd/gobgp/neighbor.go:1476 +0x34
github.com/spf13/cobra.(*Command).execute(0xc0002b4a00, {0xc0002aa9a0, 0x7, 0x7})
	/Users/liuxiang/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:856 +0x5f8
github.com/spf13/cobra.(*Command).ExecuteC(0xc00029f400)
	/Users/liuxiang/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960 +0x3ad
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/liuxiang/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:897
main.main()
	/Users/liuxiang/go/pkg/mod/github.com/osrg/gobgp@v0.0.0-20211001064702-91b91278600d/cmd/gobgp/main.go:32 +0xbe
`

--- a/cmd/gobgp/neighbor.go
+++ b/cmd/gobgp/neighbor.go
@@ -1223,8 +1223,10 @@ func modNeighbor(cmdType string, args []string) error {
 		switch cmdType {
 		case cmdAdd, cmdDel:
 			peer = &api.Peer{
-				Conf:  &api.PeerConf{},
-				State: &api.PeerState{},
+				Conf:           &api.PeerConf{},
+				State:          &api.PeerState{},
+				RouteServer:    &api.RouteServer{},
+				RouteReflector: &api.RouteReflector{},
 			}
 			if unnumbered {
 				peer.Conf.NeighborInterface = m["interface"][0]

